src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderDeleteMethodExtension.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Doctrine\QueryItemExtension;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Component\Core\Model\AdminUserInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\ShopUserInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+final class OrderDeleteMethodExtension implements QueryItemExtensionInterface
+{
+    /** @var UserContextInterface */
+    private $userContext;
+
+    public function __construct(UserContextInterface $userContext)
+    {
+        $this->userContext = $userContext;
+    }
+
+    public function applyToItem(
+        QueryBuilder $queryBuilder,
+        QueryNameGeneratorInterface $queryNameGenerator,
+        string $resourceClass,
+        array $identifiers,
+        string $operationName = null,
+        array $context = []
+    ) {
+        if (!is_a($resourceClass, OrderInterface::class, true) &&
+            $operationName !== Request::METHOD_DELETE) {
```

Comment:
I guess that here should be OR instead of AND
```suggestion
        if (
            !is_a($resourceClass, OrderInterface::class, true) ||
            $operationName !== Request::METHOD_DELETE
        ) {
```

Reply:
But what happend when `!is_a($resourceClass, OrderInterface::class, true)` return true and `$operationName !== Request::METHOD_DELETE` return false , I think that in this case better would be two `if` statements

For example if `$resourceClass` will be `ShippingMethod::class` and `$operationName` will be `Request::METHOD_GET` , it return `true || false`